modules/ServerAPI/src/server/routes/InventoryServiceRoutes.coffee

```diff
@@ -1855,6 +1856,36 @@ exports.getContainerLogsInternal = (labels, containerType, containerKind, labelT
 				response.push responseObject
 			callback response, 200
 
+exports.getContainerLogsByContainerCodes = (req, resp) ->
+	req.setTimeout 86400000
+
+	exports.getContainerLogsByContainerCodesInternal req.body, (json, statusCode) ->
+		resp.statusCode = statusCode
+		resp.json json
+
+exports.getContainerLogsByContainerCodesInternal = (containerCodes, callback) =>
+	if global.specRunnerTestmode
+		inventoryServiceTestJSON = require '../public/javascripts/spec/ServerAPI/testFixtures/InventoryServiceTestJSON.js'
+		resp.json inventoryServiceTestJSON.getContainerAndDefinitionContainerByContainerLabelInternalResponse
+	else
+		console.debug "incoming getContainerLogsByContainerCodesInternal request: '#{JSON.stringify(containerCodes)}'"
+		exports.getContainersByCodeNamesInternal(containerCodes, (containersByCodeNamesResponse, statusCode) =>
+			response = []
+			for getContainer in containersByCodeNamesResponse
+				label = getContainer.container.lsLabels[0].labelText
```

Comment:
what if the container has more than one label.  It seems like you should actually be filtering on preferred = true (should be only one preferred label)